examples/trusted_information_retrieval/database_proxy.sign

```diff
@@ -1,4 +1,8 @@
-untrusted comment: signature from minisign secret key
-RWTdfGocqvKfGVf1MpX2AnaNRZh+ov+9ZPoSdMS2C9tCqd9fmvqksLutCxI4zYy9hYqUnZf10RPAcw5GfggDCZBZfLZCegLoRwc=
-trusted comment: timestamp:1597147477	file:3606a86d9fa01d223dc3cb092c1aceb9e3afe0ed9799ac5d5b214b1610209f8f
-wMrsDIuJOM4556brM1vAVqUsbkXl/tPjsnb1SNzPmBaRlPdHteTomSXcfIgK8LUk/HeZluWBS7Y+P6YcpiF4Ag==
+-----BEGIN PUBLIC KEY-----
+f41SClNtR4i46v2Tuh1fQLbt/ZqRr1lENajCW92jyP4=
+-----END PUBLIC KEY-----
+

```

Comment:
No, I mean a separate source for the public key.

In a non-Oak context, if I were to download some app from a website, and also got both a signature and a public key from the same website, then it's tempting to verify all three together.  However, doing so doesn't really tell me anything &ndash; anyone who takes over the website can modify all three:
 - create a hacked version of the app
 - create a new keypair and serve the public key
 - use the new private key to create a new signature over the hacked app, and serve that.

For the signature to mean anything, I need to trust the public key, which means I need to acquire it by some separate path, i.e. a PKI.

Reply:
@daviddrysdale I see your point, but **I think** it does not apply here, since this is handling signatures for pieces of code, but not making any assumptions about how someone else (dat owner) would label their data with something that refers to the public key corresponding to those signatures.

This is similar to signing an Android application, and the play store checking that the signature indeed matches a public key which is bundled together with it. How someone (an end user, most likely) would get to find out about and trust such public key in the first place, is a separate problem (and one that is yet unsolved for Oak, I would say).

For now, from the data owner point of view, we expect that they magically know what public keys to trust, and would just label data accordingly, then Oak would just respect those labels.

How to make public keys discoverable, tie them to higher level identities, make sure there are no split view attacks, etc. are problems that I hope will be somewhat solved / solvable by something like key transparency / keybase, or some other PKI. Speaking of which, do you think Tink would help with any of these? (in my mind it serves a somewhat similar purpose but on the server side, i.e. handling private key rotation and matching, but maybe there is something more that it may be used for).

I also think we should look a bit more closely at how similar mechanisms work on Android, iOS and macOS, since I think there are some similarities.